report/search_incidents.go

```diff
@@ -1,397 +1,39 @@
-	// The UUID of the postmortem object attached to the incident.
-	PostmortemId *string `json:"postmortem_id,omitempty"`
-	// The monotonically increasing integer ID for the incident.
-	PublicId *int64 `json:"public_id,omitempty"`
-	// Timestamp when the incident's state was set to resolved.
-	Resolved datadog.NullableTime `json:"resolved,omitempty"`
-	// The amount of time in seconds to detect the incident. Equals the difference between `customer_impact_start` and `detected`.
-	TimeToDetect *int64 `json:"time_to_detect,omitempty"`
-	// The amount of time in seconds to call incident after detection. Equals the difference of `detected` and `created`.
-	TimeToInternalResponse *int64 `json:"time_to_internal_response,omitempty"`
-	// The amount of time in seconds to resolve customer impact after detecting the issue. Equals the difference between `customer_impact_end` and `detected`.
-	TimeToRepair *int64 `json:"time_to_repair,omitempty"`
-	// The amount of time in seconds to resolve the incident after it was created. Equals the difference between `created` and `resolved`.
-	TimeToResolve *int64 `json:"time_to_resolve,omitempty"`
-	// The title of the incident, which summarizes what happened.
-	Title string `json:"title"`
-}
-
-func (a *IncidentResponseAttributes) GetPublicId() int64 {
-	if a.PublicId != nil {
-		return *a.PublicId
-	}
-	return 0
-}
-func (a *IncidentResponseAttributes) GetFields() map[string]datadog.IncidentFieldAttributes {
-	if a.Fields != nil {
-		return *a.Fields
-	}
-	return map[string]datadog.IncidentFieldAttributes{}
-}
-
-/*
- * Execute executes the request
- * @return IncidentsSearchResponse
- */
-func searchIncidents(ctx context.Context, client *datadog.APIClient, r *searchRequest) (IncidentsSearchResponse, *nethttp.Response, error) {
-	var (
-		localVarHTTPMethod   = nethttp.MethodGet
-		localVarPostBody     interface{}
-		localVarFormFileName string
-		localVarFileName     string
-		localVarFileBytes    []byte
-		localVarReturnValue  IncidentsSearchResponse
-	)
-
-	// operationId := "SearchIncidents"
-
-	localBasePath, err := client.GetConfig().ServerURLWithContext(ctx, "IncidentsApiService.SearchIncidents")
-	if err != nil {
-		return localVarReturnValue, nil, GenericOpenAPIError{error: err.Error()}
-	}
-
-	localVarPath := localBasePath + "/api/v2/incidents/search"
-
-	localVarHeaderParams := make(map[string]string)
-	localVarQueryParams := neturl.Values{}
-	localVarFormParams := neturl.Values{}
-
+func searchIncidents(ctx context.Context, client *datadog.APIClient, r *searchRequest) (datadogV2.IncidentSearchResponse, *nethttp.Response, error) {
 	var queryOpts []string
-
 	if r.createdBefore != nil {
 		queryOpts = append(queryOpts, fmt.Sprintf("created_before:%d", *r.createdBefore))
 	}
 	if r.createdAfter != nil {
 		queryOpts = append(queryOpts, fmt.Sprintf("created_after:%d", *r.createdAfter))
 	}
 	queryOpts = append(queryOpts, r.tags...)
+	query := strings.Join(queryOpts, " ")
```

Comment:
This matches the previous implementation constructing the query: https://github.com/xornivore/incidentist/blob/33efc1adfbc0599a810e5e618c26307ee03449ba/report/search_incidents.go#L258